boards/FSM/Src/App/App_FsmWorld.c

```diff
@@ -121,3 +130,26 @@ struct RgbLedSequence *
 {
     return world->rgb_led_sequence;
 }
+
+void App_FsmWorld_RegisterSignal(struct FsmWorld *world, struct Signal *signal)
+{
+    struct SignalNode *item = malloc(sizeof(struct SignalNode));
+    assert(item != NULL);
+    item->signal = signal;
+    item->next   = NULL;
+
+    SL_APPEND(world->signals_head, item);
```

Comment:
Instead of relying on hard-coded values like `MAX_NUM_SIGNALS`, I thought it's time that we move towards using data structure from libraries to make the game more elegant.

I replaced `MAX_NUM_SIGNAL` with a header-only linked list C library: https://github.com/nbulischeck/list.h. Let me know what you think about this decision.

Reply:
I'm fine with it. Obviously we just need to be a bit more careful about properly de-allocating things in the destructor to avoid memory leaks.